third_party/random_uuid/Math.uuid.js

```diff
@@ -31,62 +31,87 @@ Dual licensed under the MIT and GPL licenses.
  *   >>> Math.uuid(8, 16) // 8 character ID (base=16)
  *   "098F4D35"
  */
-(function() {
-  // Private array of chars to use
-  var CHARS = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz'.split('');
 
-  Math.uuid = function (len, radix) {
-    var chars = CHARS, uuid = [], i;
-    radix = radix || chars.length;
+// Private array of chars to use
+const CHARS = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz'.split(
+  ''
+);
 
-    if (len) {
-      // Compact form
-      for (i = 0; i < len; i++) uuid[i] = chars[0 | Math.random()*radix];
-    } else {
-      // rfc4122, version 4 form
-      var r;
+/**
+ * Returns a random number between 0 and 1.
+ */
+export function getRandomFloat() {
+  if (crypto && crypto.getRandomValues) {
+    const arr = new Uint32Array(2);
+    crypto.getRandomValues(arr);
+    const mantissa = arr[0] * Math.pow(2, 20) + (arr[1] >>> 12);
```

Comment:
Should we extract both `Math.pow` into constants up top?

Generally, it's a bit black-magicy. I'm wondering if our live would be easier if we just based our implementation on the `Math.uuid` or `Math.uuidCompact` from that package?

Reply:
Agreed on the consts.

I'm slightly hesitant to change which of the 3 implementations we use to generate a UUID because, as you say, its black magicky and I don't understand everything going into these.  They seem to indicate that the functional output is identical so it would probably be safe?  I don't have a strong preference for any of them though.  Maybe a *slight* preference for the compact one because it reads the clearest and I don't think we are doing it enough times that we really care about performance (once per page load, maybe a few other uses here and there at a later time?)

Reply:
Though I kind of wonder (based on comments) if "minimizing calls to random() to boost performance" is causing the higher than expected intersectionality of UUIDs.

Reply:
I decided that I agreed an replaced UUIDFast with UUID.  I think the readability of uuid is much improved with my changes so it is preferable over the slower one with the regexp.